src/services/query-options.js

```diff
@@ -0,0 +1,248 @@
+      this._where.push(whereClause);
+
+      const associations = await filterParser.getAssociations(filters);
+      associations.forEach(this._neededFields.add, this._neededFields);
+    }
+  }
+
+  /**
+   * Retrict rows to those matching a search string
+   * @param {string} search search string
+   * @param {boolean|string} searchExtended if truthy, enable search in relations
+   */
+  async search(search, searchExtended) {
+    if (!search) return [];
+
+    const options = { Sequelize: this._Sequelize };
+    const fieldNames = this._requestedFields.size ? [...this._requestedFields] : null;
+    const helper = new SearchBuilder(this._model, options, { search, searchExtended }, fieldNames);
+
+    const conditions = helper.performWithSmartFields(this._options.tableAlias);
+    if (conditions) {
+      this._where.push(conditions);
+    } else {
+      this._where.push(this._Sequelize.literal('(0=1)'));
+    }
+
+    return helper.getFieldsSearched();
+  }
+
+  /**
+   * Apply a forestadmin segment
+   * @param {string} name name of the segment (from the querystring)
+   * @param {string} segmentQuery SQL query of the segment (also from querystring)
+   */
+  async segment(name) {
+    if (!name) return;
+
+    const segment = this._schema.segments?.find((s) => s.name === name);
+
+    // Segments can be provided as a sequelize scope (undocumented).
+    if (segment?.scope) {
+      this._scopes.push(segment.scope);
+    }
+
+    // ... or as a function which returns a sequelize where clause ...
+    if (typeof segment?.where === 'function') {
+      this._where.push(await segment.where());
+    }
+  }
+
+  /**
+   * Apply a segment query.
+   * FIXME Select SQL injection allows to fetch any information from database.
+   */
+  async segmentQuery(query) {
+    if (!query) return;
+
+    const [primaryKey] = _.keys(this._model.primaryKeys);
+    const queryToFilterRecords = query.trim();
+
+    new LiveQueryChecker().perform(queryToFilterRecords);
+
+    try {
+      const options = { type: this._Sequelize.QueryTypes.SELECT };
+      const records = await this._model.sequelize.query(queryToFilterRecords, options);
+      const recordIds = records.map((result) => result[primaryKey] || result.id);
+
+      this.filterByIds(recordIds);
+    } catch (error) {
+      const errorMessage = `Invalid SQL query for this Live Query segment:\n${error.message}`;
+      logger.error(errorMessage);
+      throw new ErrorHTTP422(errorMessage);
+    }
+  }
+
+  /**
+   * Apply sort instructions from a sort string in the form 'field', '-field' or 'field.subfield'.
+   * Multiple sorts are not supported
+   *
+   * @param {string} sortString a sort string
+   */
+  async sort(sortString) {
+    if (!sortString) {
+      return;
+    }
+
+    let order = 'ASC';
+    if (sortString[0] === '-') {
+      sortString = sortString.substring(1);
+      order = 'DESC';
+    }
+
+    if (sortString.indexOf('.') !== -1) {
```

Comment:
```suggestion
    if (sortString.includes('.')) {
```

Reply:
I'll change that.
I copy pasted it from the old code :)